aten/src/ATen/native/xnnpack/Convolution.cpp

```diff
@@ -79,113 +67,48 @@ bool available(
-      padding[Layout::Parameter::height],                             // input_padding_top
-      padding[Layout::Parameter::width],                              // input_padding_right
-      padding[Layout::Parameter::height],                             // input_padding_bottom
-      padding[Layout::Parameter::width],                              // input_padding_left
-      weight_nhwc.size(Layout::Filter::height),                       // kernel_height
-      weight_nhwc.size(Layout::Filter::width),                        // kernel_width
-      stride[Layout::Parameter::height],                              // subsampling_height
-      stride[Layout::Parameter::width],                               // subsampling_width
-      dilation[Layout::Parameter::height],                            // dilation_height
-      dilation[Layout::Parameter::width],                             // dilation_width
-      groups,                                                         // groups
-      weight_nhwc.size(Layout::Filter::input),                        // group_input_channels
-      weight_nhwc.size(Layout::Filter::output) / groups,              // group_output_channels
-      weight_nhwc.size(Layout::Filter::input) * groups,               // input_pixel_stride
-      weight_nhwc.size(Layout::Filter::output),                       // output_pixel_stride
-      weight_nhwc.data_ptr<float>(),                                  // kernel
-      (bias && bias->defined()) ? bias->data_ptr<float>() : nullptr,  // bias
-      output_min,                                                     // output_min
-      output_max,                                                     // output_max
-      0u,                                                             // flags
-      &convolution_op);                                               // operator
-
-  TORCH_CHECK(
-      xnn_status_success == create_status,
-      "xnn_create_convolution2d_nhwc_f32 failed!");
-
-  return Context{
-      Operator(convolution_op),
-      weight_nhwc.sizes().vec(),
-      padding,
-      stride,
-      dilation,
-  };
-}
-
 // TODO: Decouple and improve error handling and messages.
 bool usable(const Tensor& input) {
          // Input
   return (4 == input.ndimension()) &&
          (c10::DeviceType::CPU == input.device().type()) &&
          (kFloat == input.scalar_type()) &&
-         (input.size(Layout::Activation4D::batch) > 0) &&
+         (input.size(Layout::Activation4D::batch) >= 0) &&
          (input.size(Layout::Activation4D::channels) > 0) &&
          (input.size(Layout::Activation4D::height) > 0) &&
          (input.size(Layout::Activation4D::width) > 0) &&
          true;
 }
 
 Tensor run(
-    const Context& context,
+    const ContextConv2D& context,
     const Tensor& input) {
   using namespace internal;
 
   const Tensor input_nhwc = input.contiguous(MemoryFormat::ChannelsLast);
+  const Tensor padded_input_nhwc = allocate_padded_if_needed(input_nhwc);
```

Comment:
Should this be a separate diff as well?

Reply:
Yes it is possible, however I prefer we do it here, mostly for convenience reason :). However if you want I can try to separate out.